### DISCUSSION TITLE
Auto-update is possible? drawbacks?

### DISCUSSION THREAD

**Discussion by juansf99** [RavenSystem/esp-homekit-devices#1562] [General]
Hi,
With the frequency of HAA updates, I have thought it might be interesting to add an "Auto-update" feature to some of my devices. The JSON could be something like this:

{
  "c":{ 
	...
	"tz": "CET-1CEST-2,M3.5.0/2,M10.5.0/3",
	"ntp": "pool.ntp.org",
	"11": { "s": [{ "a": 2}] },
	"tt": [[11, 1, 5, 6]] 
	},
  "a":[{
    "t":.....

Its operation would make that once a week, at dawn, it enters the OTA update search mode.
Would this JSON be correct? Could there be any downside to having devices that update themselves?

**Comment by RavenSystem** [comment]
You JSON is right, and that is the way to auto-update HAA. In your case, every Saturday at 1:05h.

Of course, while device is updating, it will not be available on HomeKit, physical buttons will not work, etc.

Other issue could be that new firmware has a bug, but this case is the same if you update device by yourself.

**Comment by juansf99** [comment]
OK, thanks.
I have thought that maybe, when a major update arrives, going from v7 to v8 for example, maybe I would not be interested in updating because it could "break" the JSON due to modifying an existing function. (remember the case of service notification action "m")

Could you create a value for "a" other than 2, so that it would only update in case of minor update? Someting like this:

<html>
<body>


22 | Search for OTA firmware MINOR update (only if firmware installed via OTA method)
-- | --



</body>
</html>
  ex: If current version is 7.6.2 and the new one is 8.0.0 => DO NOT update!

 @RavenSystem, What do you think about this feature?

**Reply by RavenSystem** [reply]
Impossible. It is too complex because updates are managed by other firmware (HAA Installer), and currently it has bigger size that it can have, to let HAA to grow.

But I have not any plan changing current JSON format, because all JSON options, now are fine. And new options will not affect to current options.
